### DISCUSSION TITLE
Problem with Arduino Nano 33 IoT

### DISCUSSION THREAD

**Discussion by aregkarapetyan** [tttapa/Control-Surface#1010] [General]
I am trying to send a MIDI message over the USB using the example from the user manual. I am observing the received message in the DAW (Ableton in my case). It works for Nano 33 BLE, and I can hear the notes in the DAW. However, it doesn't work for Nano 33 IoT. It is not recognized as a MIDI device in the DAW at all. 
Is there anything I can try to make it work with the IoT board?

**Comment by tttapa** [comment]
- Which version of Control Surface are you using?
- Does it show up as a MIDI device when you try any of the examples from the MIDIUSB library? E.g. https://github.com/arduino-libraries/MIDIUSB/blob/master/examples/MIDIUSB_loop/MIDIUSB_loop.ino
- Does it show up in device manager?
- What error message do you get when compiling the following sketch?
```cpp
#include <Control_Surface.h>
// Make sure we're not silently falling back to MIDI over Serial
#ifdef CS_USB_MIDI_NOT_SUPPORTED
#error "Actual MIDI USB backend required"
#endif

USBMIDI_Interface midi;
template <class> struct PrintType;
PrintType<decltype(midi.backend)> print_type;
```

**Reply by aregkarapetyan** [reply]
- I have downloaded the latest version today (2.0.0)
- No, it doesn't show up as a MIDI device. Ableton is not recognizing it at all
- It shows up a bit weirdly in the device manager. When I connect the Nano 33 BLE (which works well), I can see a separate Serial device in "Ports." Nano 33 BLE also appears in the "Sound controllers." 
When Nano 33 IoT is connected, it shows a separate Serial device and Nano IoT 33 board separately. Nothing appears in the Sound controllers tab.
- I get the following error:
```
exit status 1
Compilation error: aggregate 'PrintType<cs::MIDIUSB_USBDeviceMIDIBackend> print_type' has incomplete type and cannot be defined
```

**Reply by tttapa** [reply]
The error message indicates that Control Surface tries to use the MIDIUSB library as its backend, which is correct. Combined with the result of the MIDIUSB example, this suggests that the issue lies with the MIDIUSB library. There are some GitHub issues discussing problems with the SAMD21 (https://github.com/arduino-libraries/MIDIUSB/issues/65, https://github.com/arduino-libraries/MIDIUSB/issues/88), but it looks like they have never been resolved :(

Unfortunately, I don't have the necessary hardware to try this myself.

For now, I'd recommend opening an issue on the MIDIUSB repository. Although it does not seem to be very active. Alternatively, you could open a thread on the Arduino forum.
